### DISCUSSION TITLE
Can I set paddleOCR to read vertical text?

### DISCUSSION THREAD

**Discussion by TLiley2** [PaddlePaddle/PaddleOCR#14463] [Q&A]
Can I set paddleOCR to read vertical text? For example.
![Hello, how are  you today_](https://github.com/user-attachments/assets/c0038432-8a17-4ba4-af81-c84fddf499db)

**Comment by GreatV** [comment]
Yes, PaddleOCR can handle vertical text, but it requires specific configuration and pre-processing to ensure accurate recognition. PaddleOCR's recognition models, like PP-OCRv4, are designed to detect and read text in multiple orientations, including rotated text, as mentioned in the linked discussions. However, for vertical text, you may need to take additional steps:

### Steps to Enable Vertical Text Recognition in PaddleOCR:
1. **Pre-process the Image:**
   - If the text is strictly vertical (e.g., characters stacked column-wise), you may need to rotate the image 90° or 270° to align the text horizontally before feeding it into the model. PaddleOCR handles horizontal text more effectively.

2. **Text Detection Model:**
   - Ensure that the detection model is capable of identifying vertical text regions. PaddleOCR's detection models (like DB or EAST) should work for vertical text, but their performance may vary depending on the dataset they were trained on.

3. **Post-processing:**
   - After recognition, if the text was rotated for processing, you can rotate it back to the original orientation.

4. **Training on Vertical Text:**
   - If vertical text is a common requirement in your use case, consider fine-tuning the recognition model on a dataset that includes vertical text. This will improve the model's accuracy for such text orientations.

### Disabling Automatic Rotation Detection (if needed):
If you find that the model is misinterpreting vertical text or rotating it incorrectly, you can disable the automatic rotation detection for 90° or 270° text as discussed [here](https://github.com/PaddlePaddle/PaddleOCR/discussions/14245). This can be done by modifying the recognition pipeline or using a custom configuration to restrict the orientations the model considers.

### Summary:
PaddleOCR can read vertical text, but for optimal results, you may need to pre-process the input (e.g., rotate the image) or fine-tune the model on vertical text datasets. If automatic rotation detection interferes, you can disable it as explained in the linked discussion.

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*
